enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -0,0 +1,225 @@
+- the ID of the corresponding OIDC client at the provider's side
+- the client secret
+- the provider's certificate authority bundle
+- any relevant extra scopes
+
+#### Authentication CR
+
+The cluster's Authentication CR (`authentication.config/cluster`) must be modified and updated with the configuration of the external OIDC provider in the `OIDCProviders` field.
+
+Once the CR gets updated, the changes will be picked up automatically by the cluster-kube-apiserver-operator, the cluster-authentication-operator and the console-operator; the operators will then update their operands accordingly and will remove all relevant components/resources.
+
+For more information on the Authentication CR API, see [API Extensions](#api-extensions).
+
+#### `oc` CLI tool
+
+`oc` supports the specification of the details of an external OIDC provider to be used via an exec plugin (currently, only `oc-oidc` is supported). See `oc login --help` for more details on how to specify these details.
+
+Note that the necessary changes have already been implemented in `oc`; see [oc#1640](https://github.com/openshift/oc/pull/1640).
+
+### API Extensions
+
+To facilitate the specification of the external OIDC provider configuration, the Authentication CRD is extended with a new field `OIDCProviders`, in its spec:
+```go
+type AuthenticationSpec struct {
+	...
+
+	// OIDCProviders are OIDC identity providers that can issue tokens
+	// for this cluster
+	// Can only be set if "Type" is set to "OIDC".
+	//
+	// At most one provider can be configured.
+	//
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=1
+	// +openshift:enable:FeatureGate=ExternalOIDC
+	OIDCProviders []OIDCProvider `json:"oidcProviders,omitempty"`
+}
+```
+
+For more details on the `OIDCProvider` type and its fields, see [here](https://github.com/openshift/api/blob/fa2f9ad8645efed0a83c24de025fd7fe791cc558/config/v1/types_authentication.go#L197).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This enhancement proposal is not relevant to Hypershift; this has been implemented independently for Hypershift (see [OCPSTRAT-933](https://issues.redhat.com/browse/OCPSTRAT-933))
+
+#### Standalone Clusters
+
+This enhancement proposal applies to standalone OCP.
+
+#### Single-node Deployments or MicroShift
+
+**SNO:** Configuring an external OIDC provider in a Single-Node deployment of OpenShift  will result in reduced resource consumption overall, due to the fact that once an external provider is configured successfully, the system will remove components and resources that are unused (e.g. the oauth-server and oauth-apiserver pods won't exist).
+
+**MicroShift:** This proposal is not relevant to MicroShift, as it does not run with multiple users.
+
+### Implementation Details/Notes/Constraints
+
+#### cluster-kube-apiserver-operator
```

Comment:
should these trigger kube-apiserver revisions or simply trigger an all-at-once instant rollout using live file reloads for the kube-apiserver?

Reply:
We've agreed that this decision must be documented in this enhancement. Putting a hold until this is resolved.

/hold